crates/router/src/workflows/payment_method_status_update.rs

```diff
@@ -0,0 +1,111 @@
+use common_utils::ext_traits::ValueExt;
+// use router_env::logger;
+use scheduler::{
+    consumer::types::process_data, utils as pt_utils, workflows::ProcessTrackerWorkflow,
+};
+
+use crate::{
+    errors,
+    logger::error,
+    routes::{metrics, SessionState},
+    types::storage::{self, PaymentMethodStatusTrackingData},
+};
+
+pub struct PaymentMethodStatusUpdateWorkflow;
+
+#[async_trait::async_trait]
+impl ProcessTrackerWorkflow<SessionState> for PaymentMethodStatusUpdateWorkflow {
+    async fn execute_workflow<'a>(
```

Comment:
nit: it would be better to have client secret expiry as a function, there might come a use case where you might need to expire client_secret somewhere else, it's better to have that code in core, and then calling it from here

Reply:
Can't be done as most operations being performed are specific to PT